src/chttpd_handlers.erl

```diff
@@ -51,6 +54,20 @@ design_handler(HandlerKey, DefaultFun) ->
 %% Internal Function Definitions
 %% ------------------------------------------------------------------
 
+is_disabled(KeyBin) ->
+    BlacklistConfig = config:get("chttpd", "url_handler_blacklist", []),
+    {ok, Blacklist} = couch_util:parse_term(BlacklistConfig),
+    Key = binary_to_list(KeyBin),
+    lists:member(Key, Blacklist).
+
+ ensure_enabled(Key) ->
+    case is_disabled(Key) of
+      true ->
+        throw("The URL endpoint " ++ Key ++ "has been disabled.");
```

Comment:
I was aiming for;

 error_info({not_found, Reason}) ->
    {404, <<"not_found">>, Reason};